corehq/motech/repeaters/models.py

```diff
@@ -690,6 +690,13 @@ class Meta:
 
     payload_generator_classes = (CaseRepeaterXMLPayloadGenerator, CaseRepeaterJsonPayloadGenerator)
 
+    def register(self, payload, fire_synchronously=False):
+        if self.repeat_records_ready.filter(payload_id=payload.get_id).count():
```

Comment:
I think this only cares if at least one exists, and does not need the exact number. It would be more efficient to use `.exists()` instead of `.count()`.